### ISSUE TITLE
Remove 'step'

### ISSUE THREAD

**Issue by oleurud** [CartoDB/Windshaft#644]
Following the https://github.com/CartoDB/Windshaft-cartodb/issues/1010 issue, we were removing step from some modules we noticed that some metrics were improving. Let's remove step to handle async code in favor of Node.js callback pattern as long as possible.

Let's go module by module with its own PR:
- [ ] lib/windshaft/backends/attributes.js
- [ ] lib/windshaft/backends/map_validator.js
- [ ] lib/windshaft/backends/preview.js
- [ ] lib/windshaft/metadata/torque-layer-metadata.js
- [ ] lib/windshaft/renderers/mapnik/factory.js
- [ ] lib/windshaft/renderers/plain/image_renderer.js
- [ ] lib/windshaft/renderers/renderer_factory.js
- [ ] lib/windshaft/renderers/torque/factory.js
- [ ] lib/windshaft/storages/mapstore.js

**Comment by oleurud** [comment]
For each Windshaft PR, we should create a new one in Windshaft-carto. Branch by branch, deploy it first in 1 server, check for no errors and then merge and deploy for every server.

**Comment by oleurud** [comment]
This task is blocked by `pg-mvt` work https://github.com/CartoDB/Windshaft/pull/646 

`Windshaft` has changes that need to make some integration code in `Windshaft-carto` (mainly in test). After talking with @Algunenano, it will be better to deploy a new version with `pg-mvt` and then, continue with this issue (this one doesn't have any priority)

**Comment by oleurud** [comment]
I remove hte blocked because I can continue with the acceptance, leaving everything ready for the deploy.

**Comment by oleurud** [comment]
## Aceptance
- [ ] [windshaft-remove-step-attributes](https://github.com/CartoDB/Windshaft-cartodb/compare/master...windshaft-remove-step-attributes) lib/windshaft/backends/attributes.js 
- [ ] [windshaft-remove-step-map-validator](https://github.com/CartoDB/Windshaft-cartodb/compare/master...windshaft-remove-step-map-validator) lib/windshaft/backends/map_validator.js
- [ ] [windshaft-remove-step-preview](https://github.com/CartoDB/Windshaft-cartodb/compare/master...windshaft-remove-step-preview) lib/windshaft/backends/preview.js
- [ ] [windshaft-remove-step-torque-layer-metadata](https://github.com/CartoDB/Windshaft-cartodb/compare/master...windshaft-remove-step-torque-layer-metadata) lib/windshaft/metadata/torque-layer-metadata.js
- [ ] [windshaft-remove-step-factory](https://github.com/CartoDB/Windshaft-cartodb/compare/master...windshaft-remove-step-factory) lib/windshaft/renderers/mapnik/factory.js
- [ ] [windshaft-remove-step-image-renderer](https://github.com/CartoDB/Windshaft-cartodb/compare/master...windshaft-remove-step-image-renderer) lib/windshaft/renderers/plain/image_renderer.js
- [ ] [windshaft-remove-step-renderer-factory](https://github.com/CartoDB/Windshaft-cartodb/compare/master...windshaft-remove-step-renderer-factory) lib/windshaft/renderers/renderer_factory.js
- [ ] [windshaft-remove-step-torque-factory](https://github.com/CartoDB/Windshaft-cartodb/compare/master...windshaft-remove-step-torque-factory) lib/windshaft/renderers/torque/factory.js
- [ ] [windshaft-remove-step-mapstore](https://github.com/CartoDB/Windshaft-cartodb/compare/master...windshaft-remove-step-mapstore) lib/windshaft/storages/mapstore.js

**Comment by oleurud** [comment]
I have discovered, I forgot a module with step!

- [ ] lib/windshaft/backends/map.js
